src/wrp_interface.c

```diff
@@ -85,8 +85,9 @@ ssize_t wi_create_response_to_message(void *data, size_t cnt, void **message)
 	    break;		
     
             default:
-            	return -1;    
-            break;
+            	response.msg_type = WRP_MSG_TYPE__AUTH;                 
+                response.u.auth.status = 501; // Not Implemented
```

Comment:
I disagree, the specification needs to be modified. Error 501 would clearly indicate to the requester that "API Is Not Implemented.", vs 400 is a general error.